### ISSUE TITLE
Add support for the <base> element

### ISSUE THREAD

**Issue by gonubana** [teamcapybara/capybara#2544]
## Meta
Capybara Version: 3.36.0
Driver Information: rack-test 1.1.0
## Expected Behavior
It should make a GET request to '/there'

## Actual Behavior
It is making a GET request to '/here/there'

## Steps to reproduce
1. Setup a basic rack app that only handles 2 request types: 'GET /here' and 'GET /there'

2. Make the following call:
```
  page.visit '/here'
```
return the following HTML:
```
  <html>
    <head>
      <base href="/">
      <title>Origin</title>
    </head>
    <body>
      <a href="there">Place</a>
    </body>
  </html>
```
and the following call:
```
  page.visit '/there'
```
return the following HTML: 
```
  <html>
    <head>
      <base href="/">
      <title>Place</title>
    </head>
    <body>
      <a href="here">Origin</a>
    </body>
  </html>
```
3. Then run the following:
```
  page.visit '/here'
  page.within_element('body') { click_link 'Place' }
```
to see it fail to return the following HTML:
```
  <html>
    <head>
      <base href="/">
      <title>Place</title>
    </head>
    <body>
      <a href="here">Origin</a>
    </body>
  </html>
```

**Comment by twalpole** [comment]
@gonubana The steps you provided don't appear to produce the result you state - the rack test driver and chrome driver have the same result, it goes to /there when clicking Place - can you provide an executable example that shows your issue?

**Comment by twalpole** [comment]
Upon further looking, this requires the initial url visited to be deeper - `/something/here` to see the issue

**Comment by twalpole** [comment]
Implemented in ec5a50f93318a1d080347f88e5d8dbfc072f1862
